compiler/rustc_builtin_macros/src/deriving/smart_ptr.rs

```diff
@@ -143,31 +151,220 @@ pub fn expand_deriving_smart_ptr(
 
     // Find the `#[pointee]` parameter and add an `Unsize<__S>` bound to it.
     let mut impl_generics = generics.clone();
+    let pointee_ty_ident = generics.params[pointee_param_idx].ident;
+    let mut self_bounds;
     {
         let p = &mut impl_generics.params[pointee_param_idx];
+        self_bounds = p.bounds.clone();
         let arg = GenericArg::Type(s_ty.clone());
         let unsize = cx.path_all(span, true, path!(span, core::marker::Unsize), vec![arg]);
         p.bounds.push(cx.trait_bound(unsize, false));
         let mut attrs = thin_vec![];
         swap(&mut p.attrs, &mut attrs);
         p.attrs = attrs.into_iter().filter(|attr| !attr.has_name(sym::pointee)).collect();
     }
+    // We should not set default values to constant generic parameters
+    // and commute bounds that indirectly involves `#[pointee]`.
```

Comment:
nit: I found "commute" a confusing word for this, "propagate" might be better? If you disagree then feel free to ignore this comment.

Reply:
I do agree that "commute" is probably not the right verb here.

Reply:
Corrected. Comments have been rephrased.